### PR TITLE
Fix inclusion issue when cache hits

### PR DESCRIPTION
**Pull request by vicancy**
@chenkennt @vwxyzh @superyyrrzz @qinezh

### CODE REVIEW

**Comment by vwxyzh** [comment]
👍

File: src/Microsoft.DocAsCode.Dfm/DocfxFlavoredIncHelper.cs [hunk 1]
```diff
@@ -39,16 +40,25 @@ private string LoadCore(IMarkdownRenderer adapter, string currentPath, string ra
                     return GenerateErrorNodeWithCommentWrapper("INCLUDE", $"Absolute path \"{currentPath}\" is not supported.", raw);
                 }
                 else
+                {
+                    // TODO: Environment.CurrentDirectory is not accurate
                     currentPath = PathUtility.MakeRelativePath(Environment.CurrentDirectory, currentPath);
```
Comment (chenkennt) [inline src/Microsoft.DocAsCode.Dfm/DocfxFlavoredIncHelper.cs]: should be docfx.json path?

**Comment by chenkennt** [comment]
>                 if (!Path.IsPathRooted(currentPath))

this will never be true?

---
Refers to: src/Microsoft.DocAsCode.Dfm/DocfxFlavoredIncHelper.cs:38 in b608d67. [](commit_id = b608d67c9a9aa7957d0dd73b74a38e4566fce929, deletion_comment = False)

File: src/Microsoft.DocAsCode.Dfm/DocfxFlavoredIncHelper.cs [hunk 2]
```diff
@@ -39,16 +40,25 @@ private string LoadCore(IMarkdownRenderer adapter, string currentPath, string ra
...
+                    if (!PathUtility.IsRelativePath(currentPath))
```
Comment (chenkennt) [inline src/Microsoft.DocAsCode.Dfm/DocfxFlavoredIncHelper.cs]: in which case will MakeRelativePath return absolute path?

File: src/Microsoft.DocAsCode.Dfm/DocfxFlavoredIncHelper.cs [hunk 3]
```diff
@@ -70,26 +81,20 @@ private string LoadCore(IMarkdownRenderer adapter, string currentPath, string ra
...
-                        UpdateHref(node, originalPath);
-
-                    result = node.WriteTo();
+                    result = UpdateToHrefFromWorkingFolder(src, currentPath);
                     result = GenerateNodeWithCommentWrapper("INCLUDE", $"Include content from \"{currentPath}\"", result);
+                    _cache.Add(currentPath, result);
                 }
+
+                // Convert back to relative path
+                return UpdateToRelativeHref(result, parent);
```
Comment (chenkennt) [inline src/Microsoft.DocAsCode.Dfm/DocfxFlavoredIncHelper.cs]: no need?

**Comment by vicancy** [comment]
@chenkennt Updated, please review

**Comment by chenkennt** [comment]
:+1:

**Pull request closed without merging**